src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -1118,23 +1118,26 @@ private FunctionGroup FirstFunctionGroupInInheritanceChain(RuntimeCore runtimeCo
             Validity.Assert(svThisPtr.IsPointer,
                             "this pointer wasn't a pointer. {89635B06-AD53-4170-ADA5-065EB2AE5858}");
 
-            int typeID = svThisPtr.metaData.type;
 
-            //Test for exact match
-            List<FunctionEndPoint> exactFeps = new List<FunctionEndPoint>();
-
-            foreach (FunctionEndPoint fep in feps)
-                if (fep.ClassOwnerIndex == typeID)
-                    exactFeps.Add(fep);
+            // We have 2 possible cases here:
+            // 1. Static method call (i.e no this pointer)
+            //    Here we will cover the specific case of static method hiding.
+            //    We do not need to check actually if the method has the "IsHideBySig" (https://docs.microsoft.com/en-us/dotnet/api/system.reflection.methodbase.ishidebysig)
+            //    because static methods can only be hidden.
+            //    In this case we simply select the function that belongs to the calling class.
+            //    The assumption here is that all function end points in "feps" have already been checked that they have the same signature.
+            // 2. Method call from an instance of a class.
+            //    Try to match with methods of that class.
+            int typeID = svThisPtr.Pointer == Constants.kInvalidIndex ? stackFrame.ClassScope : svThisPtr.metaData.type;
```

Comment:
typeId is the class ID from either the this pointer or from stackFrame;s the static func call.